### PR TITLE
LIT-135 Lightway Fragmentation

### PR DESCRIPTION
**Pull request by expressvpn-tom-l**
## Description
This PR adds Lightway Fragmentation feature. The Lightway Fragmentation automatically fragments inside packets if the size exceeds current effective pmtu.

## Motivation and Context
Improve connection quality and reduce connected-but-cannot-browse issues.

## How Has This Been Tested?
Manual tested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All active GitHub checks are passing  
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: src/he/he_internal.h [hunk 1]
```diff
@@ -125,6 +125,29 @@ struct he_ssl_ctx {
...
+typedef struct he_fragment_entry {
+  uint8_t data[HE_MAX_WIRE_MTU];
+  time_t timestamp;
+  // Linked list contains infomation of received fragments
+  he_fragment_entry_node_t *fragments;
+} he_fragment_entry_t;
+
+// Fragment table for reassembling fragments
+typedef struct he_fragment_table {
+  he_fragment_entry_t *entries[65536];
```
Comment (expressvpn-raihaan-m) [inline src/he/he_internal.h:148]: The maximum size of this is way too large for mobile

File: src/he/he_internal.h [hunk 2]
```diff
@@ -125,6 +125,29 @@ struct he_ssl_ctx {
...
 };
 
+typedef struct he_fragment_entry_node he_fragment_entry_node_t;
+
+// Information of a fragment
+typedef struct he_fragment_entry_node {
+  uint16_t begin;
+  uint16_t end;
+  bool last_frag;
+  he_fragment_entry_node_t *next;
```
Comment (expressvpn-raihaan-m) [inline src/he/he_internal.h:135]: Probably should start with the next pointer for better packing

**Review by expressvpn-raihaan-m — DISMISSED** [review]
Spoke to @expressvpn-tom-l offline about this. We will be testing this separately to ensure that we do not run into memory exhaustion scenarios on memory limited devices.

Other than that concern it overall LGTM!

File: src/he/he_internal.h (hunk 2 above)
Comment (expressvpn-tom-l) [reply src/he/he_internal.h:135]: It doesn't make a difference, because we use `pack(1)` for all these structs.

**Comment by expressvpn-tom-l** [comment]
Rebased and fixed tests.

**Review by expressvpn-raihaan-m — APPROVED** [review]

**Pull request merged**